File: app/src/main/java/com/chooloo/www/callmanager/ui/main/MainPagerAdapter.java
package com.chooloo.www.callmanager.ui.main;

import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.chooloo.www.callmanager.ui.page.PageContacts;
import com.chooloo.www.callmanager.ui.page.PageFragment;
import com.chooloo.www.callmanager.ui.page.PageRecents;

import timber.log.Timber;

public class MainPagerAdapter extends FragmentStateAdapter {

    public MainPagerAdapter(FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @Override
    public int getItemCount() {
        return 2;
    }

    @NonNull
    @Override
    public PageFragment createFragment(int position) {
        if (position == 1) {
            Timber.i("Returning recents");
            return PageRecents.newInstance();
        }
        Timber.i("Returning contacts");
        return PageContacts.newInstance();
    }

}

File: app/src/main/java/com/chooloo/www/koler/adapter/MainPagerAdapter.kt
package com.chooloo.www.koler.adapter

import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.chooloo.www.koler.ui.base.BaseActivity
import com.chooloo.www.koler.ui.contacts.ContactsFragment
import com.chooloo.www.koler.ui.recents.RecentsFragment

class MainPagerAdapter(activity: BaseActivity) : FragmentStateAdapter(activity) {
    override fun getItemCount() = 2
    override fun createFragment(position: Int): Fragment {
        return when (position) {
            1 -> RecentsFragment.newInstance(isSearchable = true)
            else -> ContactsFragment.newInstance(isSearchable = true)
        }
    }
}